### PR TITLE
[Workflow] Add task description in choosers

### PR DESCRIPTION
**Pull request by zerolab**
Adds a class method on Task (and subclasses) to allow them to provide a description.
Using it in the task choosers

### CODE REVIEW

File: wagtail/core/models.py [hunk 1]
```diff
@@ -2816,6 +2821,10 @@ def get_task_states_user_can_moderate(self, user, **kwargs):
         else:
             return TaskState.objects.none()
 
+    @classmethod
+    def get_description(cls):
+        return _("Choose which Wagtail Groups can approve this task")
```
Comment (jacobtoppm) [inline wagtail/core/models.py:2826]: I like the idea of a task description, but to me this reads more like help text/instruction for the `groups` field, unless I'm missing something. What about something like: "Members of chosen Wagtail Groups can approve this task", or "A task which can be approved by members of specific Wagtail Groups", or something along those lines?

Comment (zerolab) [reply wagtail/core/models.py]: Good shout. Updated the wording!

File: docs/advanced_topics/custom_tasks.rst [hunk 2]
```diff
@@ -195,6 +195,19 @@ For example:
...
+
+A class method that returns the human-readable description for the task.
+
+For example:
+
+.. code-block:: python
+
+    @classmethod
+    def get_description(cls):
+        return _("Choose which Wagtail Groups can approve this task")
```
Comment (jacobtoppm) [inline docs/advanced_topics/custom_tasks.rst:208]: Might be nice to update this so it's a copy of the new wording?

File: wagtail/core/models.py [hunk 3]
```diff
@@ -2816,6 +2821,10 @@ def get_task_states_user_can_moderate(self, user, **kwargs):
         else:
             return TaskState.objects.none()
 
+    @classmethod
+    def get_description(cls):
+        return _("Memebers of the chosen Wagtail Groups can approve this task")
```
Comment (jacobtoppm) [inline wagtail/core/models.py:2826]: Looks good, just one small typo - "Memebers" => "Members"

File: wagtail/core/models.py (hunk 1 above)
Comment (jacobtoppm) [reply wagtail/core/models.py]: Great, thanks!

File: docs/advanced_topics/custom_tasks.rst (hunk 2 above)
Comment (zerolab) [reply docs/advanced_topics/custom_tasks.rst]: done

**Pull request merged**